### PR TITLE
fix: correctly support includes in gitlab-ci

### LINKED ISSUE #11836: Correctly support includes in gitlab-ci
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

GitLab self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

27.14.2

### Describe the bug

I'm using the `gitlabci-include` manager and want it to update a [single configuration GitLab include](https://docs.gitlab.com/ee/ci/yaml/includes.html#include-a-single-configuration-file), like the following:

```yaml
include:
  project: org/project
  file: pipeline.yml
  ref: v1.0.0
```

However, renovate does not pick it up. I'm forced to use [*include array configuration*](https://docs.gitlab.com/ee/ci/yaml/includes.html#include-an-array-of-configuration-files) syntax in GitLab instead.

/cc @brunnerlivio

### Relevant debug logs

Logs with the single configuration include:

```
 INFO: Dependency extraction complete (repository=org/project)
       "baseBranch": "main",
       "stats": {
         "managers": {"gitlabci": {"fileCount": 1, "depCount": 2}},
         "total": {"fileCount": 1, "depCount": 2}
       }
```

Logs with the configuration array include:

```
INFO: Dependency extraction complete (repository=org/repo)
       "baseBranch": "main",
       "stats": {
         "managers": {
           "gitlabci": {"fileCount": 1, "depCount": 2},
           "gitlabci-include": {"fileCount": 1, "depCount": 1}
         },
         "total": {"fileCount": 2, "depCount": 3}
       }
```

### Have you created a minimal reproduction repository?

No reproduction repository

### PR DESCRIPTION
**Pull request by timofurrer**
Closes #11836

## Changes:

It fixes #11836 . This allows to extract single include configuration from GitLab Ci pipeline files.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/gitlabci-include/extract.spec.ts [hunk 1]
```diff
@@ -10,8 +11,17 @@ describe('manager/gitlabci-include/extract', () => {
         extractPackageFile('nothing here', '.gitlab-ci.yml', {})
       ).toBeNull();
     });
+    it('extracts single include block', () => {
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.spec.ts:14]:
```suggestion

    it('extracts single include block', () => {
```

readabillity

File: lib/manager/gitlabci-include/extract.spec.ts [hunk 2]
```diff
@@ -10,8 +11,17 @@ describe('manager/gitlabci-include/extract', () => {
...
+      const res = extractPackageFile(
+        yamlFileSingleConfig,
+        '.gitlab-ci.yml',
+        {}
+      );
+      expect(res.deps).toMatchSnapshot();
+      expect(res.deps).toHaveLength(1);
+    });
     it('extracts multiple include blocks', () => {
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.spec.ts:23]:
```suggestion

    it('extracts multiple include blocks', () => {
```

File: lib/manager/gitlabci-include/extract.ts [hunk 3]
```diff
@@ -35,15 +35,19 @@ export function extractPackageFile(
     const doc: any = load(replaceReferenceTags(content), {
       json: true,
     });
+    let includes;
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.ts:38]:
```suggestion
    if (!doc?.include) {
      return null;
    }
```

early exit when no include

File: lib/manager/gitlabci-include/extract.ts [hunk 4]
```diff
@@ -35,15 +35,22 @@ export function extractPackageFile(
     const doc: any = load(replaceReferenceTags(content), {
       json: true,
     });
+    if (!doc?.include) {
+      return null;
+    }
     if (doc?.include && is.array(doc.include)) {
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.ts:41]:
```suggestion
```

File: lib/manager/gitlabci-include/extract.ts [hunk 5]
```diff
@@ -35,15 +35,22 @@ export function extractPackageFile(
...
-        if (includeObj.file && includeObj.project) {
-          const dep = extractDepFromIncludeFile(includeObj);
-          if (config.endpoint) {
-            dep.registryUrls = [config.endpoint.replace(/\/api\/v4\/?/, '')];
-          }
-          deps.push(dep);
+      includes = doc.include;
+    } else {
+      includes = [doc.include];
+    }
```
Comment (viceice) [inline lib/manager/gitlabci-include/extract.ts:45]:
```suggestion
```

**Review by viceice — DISMISSED** [review]

**Comment by timofurrer** [comment]
@viceice I noticed that the coverage dropped -0.01% ... I assume you are targeting 100% and therefore you cannot merge it yet. Is that correct?

**Comment by viceice** [comment]
Yes, check the local coverage folder, there are html files to review the covered code

**Review by viceice — APPROVED** [review]

**Pull request merged**